phantazm-neuron-minestom/src/main/java/com/github/phantazmnetwork/neuron/bindings/minestom/entity/ContextualSpawner.java

```diff
@@ -0,0 +1,45 @@
+package com.github.phantazmnetwork.neuron.bindings.minestom.entity;
+
+import com.github.phantazmnetwork.neuron.bindings.minestom.ContextProvider;
+import net.minestom.server.coordinate.Point;
+import net.minestom.server.instance.Instance;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+/**
+ * A {@link Spawner} implementation that uses a single {@link ContextProvider} to build and spawn all entities.
+ */
+@SuppressWarnings("ClassCanBeRecord")
+public class ContextualSpawner implements Spawner {
+    private final ContextProvider contextProvider;
+
+    /**
+     * Creates a new ContextualSpawner that will use the given {@link ContextProvider} to build all entities.
+     * @param contextProvider the ContextProvider which will be used to construct new entities
+     */
+    public ContextualSpawner(@NotNull ContextProvider contextProvider) {
+        this.contextProvider = Objects.requireNonNull(contextProvider, "contextProvider");
+    }
+
+    @Override
+    public <TDescriptor extends MinestomDescriptor, TEntity extends NeuralEntity> @NotNull TEntity spawnEntity(
+            @NotNull Instance instance, @Nullable Point point, @NotNull TDescriptor type,
```

Comment:
when would we spawn something without a spawn location? kind of prefer having a concrete loc

Reply:
sure